src/charm.py

```diff
@@ -23,14 +42,43 @@
 
 
 class LMAProxyCharm(CharmBase):
+    """This class instantiates Charmed Operator libraries and sets the status of the charm.
+
+    No actual work is performed by the charm payload. It is only the libraries which are
+    expected to listen on relations, respond to relation events, and set outgoing relation
+    data.
+    """
 
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
 
-        self._stored.set_default(have_prometheus=False)
-        self._stored.set_default(have_targets=False)
+        self._stored.set_default(
+            have_grafana=False, have_dashboards=False, have_prometheus=False, have_targets=False
```

Comment:
It might run HA in hot/cold mode. Ultimately the non-leader units should do all the work tge leader does, except writing relation data bags.

Reply:
It also has absolutely no "actual" workload by itself.

In practice, all it's doing is mutating data which comes from reactive charms and passing it along. Since reactive charms actually send it with a an RPC-like mechanism as individual requests rather than just sticking it on the bucket, there's not even a big risk there.

The only thing this values are used for is setting the status, so users have immediate visual feedback if they'd provided a Prometheus relation and no targets, or Grafana and no dashboards, so they can tell from `juju status` rather than the logs why nothing's passed through.